In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

# Load the dataset
data = pd.read_csv('/workspaces/codespaces-jupyter/data/Copy of loan.csv')
data.shape

(614, 13)

In [2]:
data.info()

data['Gender'] = data['Gender'].replace('Male', 1).replace(['Female', 'Other'], 0)



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    float64
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(5), int64(1), object(7)
memory usage: 62.5+ KB


In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    float64
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    float64
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(6), int64(1), object(6)
memory usage: 62.5+ KB


In [4]:
data['Loan_Status'] = data['Loan_Status'].replace('Y', 1).replace(['N'], 0)
loan_status_items = data['Loan_Status'].unique()

print(loan_status_items)


[0 1]


In [5]:
data['Education'] = data['Education'].replace('Graduate', 1).replace(['Not Graduate'], 0)
education_status_items = data['Education'].unique()

print(education_status_items)



[0 1]


In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    float64
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    float64
 4   Education          614 non-null    int64  
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    int64  
dtypes: float64(6), int64(3), object(4)
memory usage: 62.5+ KB


In [7]:
data = data.dropna()

In [8]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error


# Select the columns you want to use as features
features = ['Gender', 'Dependents', 'Education', 'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
       'Loan_Amount_Term', 'Credit_History']

# Select the column you want to predict
target = 'Loan_Status'

# Split the data into training and test sets
X_train, X_test, y_train, y_test = train_test_split(data[features], data[target], test_size=0.2)

# Create and fit the model
model = LinearRegression()
model.fit(X_train, y_train)

# Make predictions on the test set
y_pred = model.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, y_pred)
print(f'Mean squared error: {mse:.2f}')

# Show the coefficients for each feature
coefficients = pd.DataFrame({'Feature': features, 'Coefficient': model.coef_})
print(coefficients)

Mean squared error: 0.18
             Feature  Coefficient
0             Gender     0.078357
1         Dependents     0.020414
2          Education     0.110643
3    ApplicantIncome    -0.000003
4  CoapplicantIncome    -0.000010
5         LoanAmount    -0.000494
6   Loan_Amount_Term    -0.000131
7     Credit_History     0.704743


The output you've provided is the result of running a linear regression model on a dataset. The model is predicting the 'Loan_Status' based on various features like 'Gender', 'Dependents', 'Education', etc.

The 'Mean squared error' is a measure of how well the model is performing, with lower values indicating better performance. In this case, the MSE is 0.18.

The table below the MSE is showing the coefficients of each feature in the linear regression model. These coefficients represent the relationship between each feature and the target variable 'Loan_Status'. For example, the coefficient for 'Gender' is 5.841470e-02, which means for each unit increase in 'Gender', the 'Loan_Status' increases by 5.841470e-02 units, holding all other features constant.

However, it seems like you're asking for an explanation of Markdown in the context of this Python code. Markdown is not directly relevant to this Python code. Markdown is a lightweight markup language for creating formatted text. It's often used for writing rich text on the web and in text editors. In a Jupyter notebook, you can use Markdown to write explanatory text, add headings, create lists, and so on. But in the Python code you've provided, there's no Markdown being used.

In [9]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression, Ridge, Lasso, ElasticNet
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor, GradientBoostingRegressor
from sklearn.metrics import mean_absolute_error, mean_squared_error


# Select the columns you want to use as features
features = ['Gender', 'Dependents', 'Education', 'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
       'Loan_Amount_Term', 'Credit_History']

# Select the column you want to predict
target = 'Loan_Status'

# Split the data into training and test sets
X_train, X_test, y_train, y_test = train_test_split(data[features], data[target], test_size=0.2)

# Define the models to evaluate
models = {
    'Linear Regression': LinearRegression(),
    'Ridge Regression': Ridge(),
    'Lasso Regression': Lasso(),
    'Elastic Net Regression': ElasticNet(),
    'Decision Tree Regression': DecisionTreeRegressor(),
    'Random Forest Regression': RandomForestRegressor(),
    'Gradient Boosting Regression': GradientBoostingRegressor()
}

# Fit and evaluate each model
for name, model in models.items():
    # Fit the model
    model.fit(X_train, y_train)

    # Make predictions on the test set
    y_pred = model.predict(X_test)

    # Calculate the mean absolute error (MAE)
    mae = mean_absolute_error(y_test, y_pred)

    # Calculate the root mean squared error (RMSE)
    mse = (mean_squared_error(y_test, y_pred))

    # Print the results
    print(f'{name}:')
    print(f'  MAE: {mae:.2f}')
    print(f'  MSE: {mse:.2f}')

Linear Regression:
  MAE: 0.32
  MSE: 0.17
Ridge Regression:
  MAE: 0.32
  MSE: 0.17
Lasso Regression:
  MAE: 0.43
  MSE: 0.21
Elastic Net Regression:
  MAE: 0.43
  MSE: 0.21
Decision Tree Regression:
  MAE: 0.33
  MSE: 0.33
Random Forest Regression:
  MAE: 0.30
  MSE: 0.18
Gradient Boosting Regression:
  MAE: 0.32
  MSE: 0.19


This output is showing the performance of various regression models on a dataset. The performance is measured using two metrics: Mean Absolute Error (MAE) and Mean Squared Error (MSE).

Mean Absolute Error (MAE): This is the average of the absolute differences between the predicted and actual values. It gives an idea of how wrong the predictions were. The measure gives an idea of the magnitude of the error, but no idea of the direction (e.g. over or under predicting).

Mean Squared Error (MSE): This is the average of the squared differences between the predicted and actual values. It is a popular metric because squaring the errors penalizes larger errors more than smaller ones.

Each model (Linear Regression, Ridge Regression, Lasso Regression, etc.) has its own MAE and MSE values. Lower values for both metrics indicate better model performance.

For example, the Random Forest Regression model has an MAE of 0.30 and an MSE of 0.18, which means on average, the model's predictions are 0.30 units away from the actual values (MAE), and the average squared difference between the predicted and actual values is 0.18 (MSE). This model seems to perform better than others based on these metrics.